python/src/equistore/operations/reduce_over_samples.py

```diff
@@ -101,69 +101,94 @@ def _reduce_over_samples_block(
     return result_block
 
 
-def sum_over_samples(tensor: TensorMap, sample_names: List[str]) -> TensorMap:
+def _reduce_over_samples(
+    tensor: TensorMap, samples_names: Union[List[str], str], reduction: str
+) -> TensorMap:
     """Create a new :py:class:`TensorMap` with the same keys as
     as the input ``tensor``, and each :py:class:`TensorBlock` is obtained
-    summing the corresponding input :py:class:`TensorBlock` over the rows with
-    the same ``sample_names``.
+    summing the corresponding input :py:class:`TensorBlock` over the ``samples_names``
+    indeces.
 
-    For example if ``sample_name = ["structure"]``, the function sums over all
-    the sample with the same structure, and if ``sample_name = ["structure",
-    "center"]`` it sums over all the rows with the same values for both
-    ``"structure"`` and ``"center"`` samples.
+    Both "sum" and "mean" reductions can be performed.
+
+    ``samples_names`` tells over which indeces of the sparse matrix the reduction
+    is performed. It accept both a single string or a list of the string
+    with the sample_names corresponding to the directions along which the
+    reduction is performed.
+    Moreover if only one sample name is given both a single string or a list with
+    a string is allowed: ``samples_names = ["center"]`` or equivalently
+    ``samples_names = "center"``.
```

Comment:
I would not duplicate the documentation here, since this is a private function, and I would rather have fewer places where we need to update documentation